File: python/balance_parens.py
def balance_parens(par) :
    stack=[]
    split_str=list(par)
    for i in range(len(par)):
        if par[i]=='(':
            stack.append(i)
        elif par[i]==')':
            if len(stack) !=0:
                    stack.pop()
            else:
                split_str[i]=""
    for i in stack:
        split_str[i]=""
    return '' .join(split_str)


print(balance_parens("a(b)c)"))
print(balance_parens('()'))
print(balance_parens(')('))

File: js/balanceParens.js
balanceParens = (str) => {
  const res = [];
  const splitStr = str.split("");
  for (let i = 0; i < str.length; i++) {
    const item = str[i];
    if (item === "(") res.push(i);
    else if (item === ")") {
      if (res.length === 0) {
        splitStr[i] = "";
      } else {
        res.pop();
      }
    }
  }
  for (let i = 0; i < res.length; i++) {
    const item = res[i];
    splitStr[item] = "";
  }

  return splitStr.join("");
};

console.log(balanceParens("a(b)c)"));
console.log(balanceParens("()"));
console.log(balanceParens(")("));
// module.exports = { balanceParens };
